packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTable/TransactionsTable.tsx

```diff
@@ -0,0 +1,250 @@
+
+    // start identifying newer transactions
+    const newerTransactions: MergedTransaction[] = []
+    locallyStoredTransactions?.forEach(tx => {
+      // only check deposit txns if you're on deposits' tab, and vice-versa
+      const isMatchingTab =
+        (isDeposit(tx) && type === 'deposits') ||
+        (isWithdrawal(tx) && type === 'withdrawals')
+      if (!isMatchingTab) return
+
+      // check if the local-state tx is newer than the latest subgraph date
+      const pendingTxCreationDate = dayjs(tx.createdAt)
+      const isTxAfterSubgraphTxs =
+        pendingTxCreationDate.isAfter(firstSubgraphTransactionTimestamp) &&
+        tx.createdAt !== subgraphTransactions[0]?.['createdAt']
+
+      if (isTxAfterSubgraphTxs) {
+        newerTransactions.push(tx)
+      }
+    })
+
+    // if newer txns found, append it to the existing subgraph transactions
+    return [...newerTransactions.reverse(), ...subgraphTransactions]
+  }, [transactions, localTransactionsKey])
+
+  const pendingTransactionsMap = useMemo(() => {
+    // map of all the locally-stored pending transactions
+    // so that tx rows can easily subscribe to live-local status without refetching table data
+    const pendingTxMap = new Map<string, MergedTransaction>()
+    pendingTransactions.forEach(tx => {
+      pendingTxMap.set(tx.txId, tx)
+    })
+    return pendingTxMap
+  }, [localTransactionsKey])
+
+  const status = (() => {
+    if (loading) return TableStatus.LOADING
+    if (error) return TableStatus.ERROR
+    return TableStatus.SUCCESS
+  })()
+
+  const noSearchResults = !!(
+    pageParams?.searchString?.length &&
+    status === TableStatus.SUCCESS &&
+    !transactions?.length
+  )
+
+  return (
+    <>
+      {/* search and pagination buttons */}
+      <TableActionHeader
+        type={type}
+        pageParams={pageParams}
+        setPageParams={setPageParams}
+        transactions={transactions}
+        loading={loading}
+      />
+
+      {
+        <table className="w-full overflow-hidden  rounded-b-lg bg-white">
+          <thead className="text-left text-sm text-gray-10">
+            <tr>
+              <th className="py-3 pl-6 pr-3 font-normal">Status</th>
+              <th className="px-3 py-3 font-normal">Time</th>
+              <th className="px-3 py-3 font-normal">Amount</th>
+              <th className="px-3 py-3 font-normal">TxID</th>
+              <th className="py-3 pl-3 pr-6 font-normal">
+                {/* Empty header text */}
+              </th>
+            </tr>
+          </thead>
+
+          <tbody>
+            {status === TableStatus.LOADING && <TableBodyLoading />}
+
+            {status === TableStatus.ERROR && <TableBodyError />}
+
+            {/* when there are no search results found */}
+            {status === TableStatus.SUCCESS && noSearchResults && (
+              <NoDataOverlay />
+            )}
+
+            {status === TableStatus.SUCCESS && !noSearchResults && (
+              <>
```

Comment:
very minor but this is an unnecessary React Fragment.
`( <> {` can all be removed

Reply:
Simplified this.